### PR TITLE
Added option to construct all layers at once.

### PR DESCRIPTION
**Pull request by dppdppd**
Major:
- Added option to construct all layers at once by specifying Active_Layer = 0

Minor:
- Automated Resolution to be dependent on $preview

### CODE REVIEW

File: IKEA_Samla_Inserts.scad [hunk 1]
```diff
@@ -343,17 +342,24 @@ module Create_Samla_Insert(width, depth, height, scale_width, scale_depth, width
...
+        translate( [ 0, layer * 11_depth * 11_scale_depth,0] )
+            Create_Samla_Insert(layer, 11_width, 11_depth, 11_height, 11_scale_width, 11_scale_depth, 11_width_handle, 11_depth_handle, 11_width_cutout, 11_depth_cutout, 11_scale_handle, 11_scale_cutout, 11_handle_cutout_height, 11_diameter, 11_diameter2, Resolution);
     }
     else if (Box_Size == "22")
     {
-        Create_Samla_Insert(22_width, 22_depth, 22_height, 22_scale_width, 22_scale_depth, 22_width_handle, 22_depth_handle, 22_width_cutout, 22_depth_cutout, 22_scale_handle, 22_scale_cutout, 22_handle_cutout_height, 22_diameter, 22_diameter2, Resolution);
+        translate( [ 0, layer * 22_depth * 22_scale_depth, 0,0] )
+            Create_Samla_Insert(layer, 22_width, 22_depth, 22_height, 22_scale_width, 22_scale_depth, 22_width_handle, 22_depth_handle, 22_width_cutout, 22_depth_cutout, 22_scale_handle, 22_scale_cutout, 22_handle_cutout_height, 22_diameter, 22_diameter2, Resolution);
```
Comment (Alex9779) [inline IKEA_Samla_Inserts.scad]: I'd rather like to see here "(layer-1)" to center the first layer at the origin as it was before...

**Comment by Alex9779** [comment]
nice, didn't know that that thing with the resolution is possible...
Just two things, one thing I commented in the code, other is could you do a rebase of your changes instead of a merge of my changes since you forked?

**Comment by dppdppd** [comment]
Hi there. So I fixed the centering and also fixed a bug I created where it wouldn't re-center to origin when specifying a layer other than 0 or 1. Anyways, I think I rebased and I think I committed correctly but I'm kinda just mashing the buttons at this point. Let me know if I'm close.

Cheers.

**Comment by Alex9779** [comment]
Ok, well unfortunately is the history still not nice, so I think I will just cherry pick your changes...

**Comment by Alex9779** [comment]
Ok the history is not nice but I wasn't able to cherry pick also in a nice way so I just merged...

For the future, use "rebase" if you want to update your branch to upstream changes...

**Comment by dppdppd** [comment]
Sorry for the hassle. Glad I could contribute after all.  Thanks for the project!

**Pull request merged**